src/middlewared/middlewared/service/decorators.py

```diff
@@ -48,6 +51,26 @@ def filterable_internal(fn):
     return filterable_internal
 
 
+def filterable_api_method(fn=None, /, *, roles=None, item=None, private=True):
+    def filterable_internal(fn):
+        fn._filterable = True
+        if hasattr(fn, 'wraps'):
+            fn.wraps._filterable = True
+
+        if item:
+            returns = query_result(item)
+        else:
+            returns = GenericQueryResult
```

Comment:
We should not allow this, we should introduce schemas for everything. Or is it impossible in this case?

Reply:
I envision this generally being used in private APIs. We have large amount of private usage of `filter_list` that can benefit from the normalization and validation. I don't see benefit to providing versioning for private APIs (it is in fact probably harmful in long-run).

I can make it so that `item` is required if non-private.

Reply:
Excellent idea, let’s do that!